### PR TITLE
Execute predicates only for similar pods.

### PR DESCRIPTION
**Pull request by krzysztof-jastrzebski**

### CODE REVIEW

File: cluster-autoscaler/core/utils.go [hunk 1]
```diff
@@ -174,6 +174,34 @@ func FilterOutExpendablePods(pods []*apiv1.Pod, expendablePodsPriorityCutoff int
...
+			} else {
+				glog.V(2).Infof("Predicate failed: %v", err)
+				schedulable = false
+				podSchedulable.set(pod, false)
+			}
+		}
+		if schedulable {
+			schedulablePods = append(schedulablePods, pod)
+		} else {
+			glogx.V(2).UpTo(loggingQuota).Infof("Pod %s can't be scheduled on %s", pod.Name, nodeGroupId)
```
Comment (MaciekPytel) [inline cluster-autoscaler/core/utils.go]: I don't like logging this line regardless if we actually run predicate or not. If we did run predicates for this pod it should be obvious from the log above (if it's not we should update that log). Also if we hit the limit some pods will have 2 lines ('Predicate failed' and 'cant be scheduled') and some will only have the former.

Maybe just change the `else` to `else if found` and update the log messages accordingly?

**Comment by MaciekPytel** [comment]
Looks much nicer than I thought it would when we talked about it, very cool :)

That being said it's a non-trivial change in core logic. Can you run a quick sanity check and post a relevant part of the log (or just description of test)?

File: cluster-autoscaler/core/utils.go (hunk 1 above)
Comment (krzysztof-jastrzebski) [reply cluster-autoscaler/core/utils.go]: done

**Comment by krzysztof-jastrzebski** [comment]
Log:
I0529 07:31:39.372369      11 utils.go:197] Pod gputest-bb6bccf57-d8fjd can't be scheduled on https://content.googleapis.com/compute/v1/projects/project/zones/us-central1-c/instanceGroups/nap-n1-standard-8-1527579099-temporary-mig, predicate failed: PodFitsResources predicate mismatch, cannot put default/gputest-bb6bccf57-d8fjd on nap-n1-standard-8-1527579099-temporary-mig-autoprovisioned-template-7420029401873005924, reason: Insufficient nvidia.com/gpu
I0529 07:31:39.372563      11 utils.go:188] Pod gputest-bb6bccf57-89rlr can't be scheduled on https://content.googleapis.com/compute/v1/projects/project/zones/us-central1-c/instanceGroups/nap-n1-standard-8-1527579099-temporary-mig. Used cached predicate check results
I0529 07:31:39.373818      11 utils.go:188] Pod gputest-bb6bccf57-stfc5 can't be scheduled on https://content.googleapis.com/compute/v1/projects/project/zones/us-central1-c/instanceGroups/nap-n1-standard-8-1527579099-temporary-mig. Used cached predicate check results
I0529 07:31:39.374110      11 utils.go:188] Pod gputest-bb6bccf57-z42bm can't be scheduled on https://content.googleapis.com/compute/v1/projects/project/zones/us-central1-c/instanceGroups/nap-n1-standard-8-1527579099-temporary-mig. Used cached predicate check results
I0529 07:31:39.374171      11 utils.go:188] Pod gputest-bb6bccf57-xsprn can't be scheduled on https://content.googleapis.com/compute/v1/projects/project/zones/us-central1-c/instanceGroups/nap-n1-standard-8-1527579099-temporary-mig. Used cached predicate check results
I0529 07:31:39.374687      11 utils.go:188] Pod gputest-bb6bccf57-75z4c can't be scheduled on https://content.googleapis.com/compute/v1/projects/project/zones/us-central1-c/instanceGroups/nap-n1-standard-8-1527579099-temporary-mig. Used cached predicate check results
I0529 07:31:39.374936      11 utils.go:188] Pod gputest-bb6bccf57-8whr6 can't be scheduled on https://content.googleapis.com/compute/v1/projects/project/zones/us-central1-c/instanceGroups/nap-n1-standard-8-1527579099-temporary-mig. Used cached predicate check results
I0529 07:31:39.375833      11 utils.go:188] Pod gputest-bb6bccf57-mkvxr can't be scheduled on https://content.googleapis.com/compute/v1/projects/project/zones/us-central1-c/instanceGroups/nap-n1-standard-8-1527579099-temporary-mig. Used cached predicate check results
I0529 07:31:39.375904      11 utils.go:188] Pod gputest-bb6bccf57-dblvn can't be scheduled on https://content.googleapis.com/compute/v1/projects/project/zones/us-central1-c/instanceGroups/nap-n1-standard-8-1527579099-temporary-mig. Used cached predicate check results
I0529 07:31:39.376351      11 utils.go:188] Pod gputest-bb6bccf57-cbhn7 can't be scheduled on https://content.googleapis.com/compute/v1/projects/project/zones/us-central1-c/instanceGroups/nap-n1-standard-8-1527579099-temporary-mig. Used cached predicate check results
I0529 07:31:39.376694      11 utils.go:188] Pod gputest-bb6bccf57-ddqdh can't be scheduled on https://content.googleapis.com/compute/v1/projects/project/zones/us-central1-c/instanceGroups/nap-n1-standard-8-1527579099-temporary-mig. Used cached predicate check results
I0529 07:31:39.378244      11 utils.go:188] Pod gputest-bb6bccf57-f6v2j can't be scheduled on https://content.googleapis.com/compute/v1/projects/project/zones/us-central1-c/instanceGroups/nap-n1-standard-8-1527579099-temporary-mig. Used cached predicate check results
I0529 07:31:39.378386      11 utils.go:188] Pod gputest-bb6bccf57-vht9n can't be scheduled on https://content.googleapis.com/compute/v1/projects/project/zones/us-central1-c/instanceGroups/nap-n1-standard-8-1527579099-temporary-mig. Used cached predicate check results
I0529 07:31:39.379115      11 utils.go:188] Pod gputest-bb6bccf57-s7czq can't be scheduled on https://content.googleapis.com/compute/v1/projects/project/zones/us-central1-c/instanceGroups/nap-n1-standard-8-1527579099-temporary-mig. Used cached predicate check results
I0529 07:31:39.380479      11 utils.go:188] Pod gputest-bb6bccf57-ttkft can't be scheduled on https://content.googleapis.com/compute/v1/projects/project/zones/us-central1-c/instanceGroups/nap-n1-standard-8-1527579099-temporary-mig. Used cached predicate check results
I0529 07:31:39.380890      11 utils.go:188] Pod gputest-bb6bccf57-g7tw2 can't be scheduled on https://content.googleapis.com/compute/v1/projects/project/zones/us-central1-c/instanceGroups/nap-n1-standard-8-1527579099-temporary-mig. Used cached predicate check results
I0529 07:31:39.381133      11 utils.go:188] Pod gputest-bb6bccf57-rfkjl can't be scheduled on https://content.googleapis.com/compute/v1/projects/project/zones/us-central1-c/instanceGroups/nap-n1-standard-8-1527579099-temporary-mig. Used cached predicate check results
I0529 07:31:39.381383      11 utils.go:188] Pod gputest-bb6bccf57-c6467 can't be scheduled on https://content.googleapis.com/compute/v1/projects/project/zones/us-central1-c/instanceGroups/nap-n1-standard-8-1527579099-temporary-mig. Used cached predicate check results
I0529 07:31:39.381674      11 utils.go:188] Pod gputest-bb6bccf57-kthjq can't be scheduled on https://content.googleapis.com/compute/v1/projects/project/zones/us-central1-c/instanceGroups/nap-n1-standard-8-1527579099-temporary-mig. Used cached predicate check results
I0529 07:31:39.381741      11 utils.go:188] Pod gputest-bb6bccf57-dvdvm can't be scheduled on https://content.googleapis.com/compute/v1/projects/project/zones/us-central1-c/instanceGroups/nap-n1-standard-8-1527579099-temporary-mig. Used cached predicate check results
I0529 07:31:39.381801      11 utils.go:188] Pod gputest-bb6bccf57-g5c22 can't be scheduled on https://content.googleapis.com/compute/v1/projects/project/zones/us-central1-c/instanceGroups/nap-n1-standard-8-1527579099-temporary-mig. Used cached predicate check results
I0529 07:31:39.421201      11 utils.go:203] 79 other pods can't be scheduled on &NodeInfo{Pods:[kube-proxy-https://content.googleapis.com/compute/v1/projects/project/zones/us-central1-c/instanceGroups/nap-n1-standard-8-1527579099-temporary-mig-968086288776002427], Requ

**Comment by krzysztof-jastrzebski** [comment]
I have fixed this.

**Pull request merged**